PermissionScope/Structs.swift

```diff
@@ -22,6 +53,8 @@ import Foundation
     }
     
     public var description: String {
+        /* TODO: This will not compile when used in a project (unless project is asking ALL permissions) because any permission that is not used will have it's enum undefined due to the #ifendif statements around the enum definition above.
+        */
```

Comment:
Since this doesn't compile, is the goal to address that issue before merging this PR?  Or merge as-is? I usually think it's best to avoid merging into master with commits that don't compile, since that can cause headaches for things like `bisect`.

Also do we have a workaround for this? The multiple `if` pattern that Ying suggested?